{"id":"33855001546","type":"IssuesEvent","actor":{"id":57787676,"login":"baentsch","display_login":"baentsch","gravatar_id":"","url":"https://api.github.com/users/baentsch","avatar_url":"https://avatars.githubusercontent.com/u/57787676?"},"repo":{"id":179335992,"name":"open-quantum-safe/boringssl","url":"https://api.github.com/repos/open-quantum-safe/boringssl"},"payload":{"action":"closed","issue":{"url":"https://api.github.com/repos/open-quantum-safe/boringssl/issues/102","repository_url":"https://api.github.com/repos/open-quantum-safe/boringssl","labels_url":"https://api.github.com/repos/open-quantum-safe/boringssl/issues/102/labels{/name}","comments_url":"https://api.github.com/repos/open-quantum-safe/boringssl/issues/102/comments","events_url":"https://api.github.com/repos/open-quantum-safe/boringssl/issues/102/events","html_url":"https://github.com/open-quantum-safe/boringssl/issues/102","id":1775940865,"node_id":"I_kwDOCrBzOM5p2rUB","number":102,"title":"Windows/Mac Interop Test Server Handshake Tests Failing","user":{"login":"qikdauie","id":63803208,"node_id":"MDQ6VXNlcjYzODAzMjA4","avatar_url":"https://avatars.githubusercontent.com/u/63803208?v=4","gravatar_id":"","url":"https://api.github.com/users/qikdauie","html_url":"https://github.com/qikdauie","followers_url":"https://api.github.com/users/qikdauie/followers","following_url":"https://api.github.com/users/qikdauie/following{/other_user}","gists_url":"https://api.github.com/users/qikdauie/gists{/gist_id}","starred_url":"https://api.github.com/users/qikdauie/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/qikdauie/subscriptions","organizations_url":"https://api.github.com/users/qikdauie/orgs","repos_url":"https://api.github.com/users/qikdauie/repos","events_url":"https://api.github.com/users/qikdauie/events{/privacy}","received_events_url":"https://api.github.com/users/qikdauie/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2023-06-27T01:39:56Z","updated_at":"2023-12-04T07:24:55Z","closed_at":"2023-12-04T07:24:54Z","author_association":"NONE","active_lock_reason":null,"body":"Howdy there, everyone. I am currently trying to build the most recent snapshot tag on this repository [91efd57], with liboqs 0.8.0 [db08f12]. I am able to build and integrate both repositories together; however, when testing I am getting a lot of issues with OQS Handshakes.\r\n\r\nWhen running `ninja run_tests` in boringssl's build directory, I receive many failed tests that look like this for all of the NIDs added by OQS:\r\n```\r\n[----------] 1 test from WithSignatureNIDs/OQSHandshakeTest\r\n[ RUN      ] WithSignatureNIDs/OQSHandshakeTest.AllKemSignatureTests/1\r\nNID of group under test: 965\r\nNID of group under test: 966\r\nNID of group under test: 977\r\nNID of group under test: 978\r\nNID of group under test: 979\r\nNID of group under test: 980\r\nNID of group under test: 981\r\nNID of group under test: 982\r\nNID of group under test: 983\r\nNID of group under test: 984\r\nNID of group under test: 985\r\nNID of group under test: 986\r\nNID of group under test: 1211\r\nunknown file: error: SEH exception with code 0xc0000005 thrown in the test body.\r\n\r\n[  FAILED  ] WithSignatureNIDs/OQSHandshakeTest.AllKemSignatureTests/1, where GetParam() = 1143 (2997 ms)\r\n[----------] 1 test from WithSignatureNIDs/OQSHandshakeTest (2997 ms total)\r\n```\r\n \r\n \r\n When running `pytest oqs_scripts\\test_with_interop_server.py` im getting: `410 failed, 94 passed in 198.41s (0:03:18)`, where excerpts of the test's output look like this:\r\n\r\n```\r\n________________________________ test_sig_kex_pair[sphincsshake256ssimple-p521_hqc256] ________________________________\r\n\r\nsig = 'sphincsshake256ssimple', kex = 'p521_hqc256', bssl = 'build\\\\tool\\\\bssl'\r\nserver_CA_cert = <tempfile._TemporaryFileWrapper object at 0x000001C778E2E550>\r\nserver_port_assignments = {'dilithium2': {'*': 6086, 'bikel1': 6087, 'frodo640aes': 6088, 'frodo640shake': 6089, ...}, 'dilithium3': {'*': 6104,...6122, 'frodo1344shake': 6123, ...}, 'ecdsap256': {'*': 6000, 'bikel1': 6001, 'bikel3': 6002, 'bikel5': 6003, ...}, ...}\r\n\r\n    @pytest.mark.parametrize('kex', kexs)\r\n    @pytest.mark.parametrize('sig', sigs)\r\n    def test_sig_kex_pair(sig, kex, bssl, server_CA_cert, server_port_assignments):\r\n        if kex == 'prime256v1':\r\n           server_port = server_port_assignments[sig][\"*\"]\r\n        elif kex == 'x25519':\r\n           server_port = server_port_assignments[sig]['X25519']\r\n        else:\r\n>          server_port = server_port_assignments[sig][kex]\r\nE          KeyError: 'sphincsshake256ssimple'\r\n```\r\n\r\nand below in the Summary, the top looks like this:\r\n\r\n```\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[ecdsap256-x25519] - KeyError: 'X25519'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[ecdsap256-bikel1] - AssertionError: Connecting to 158.177.128.14:6001\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[ecdsap256-p256_bikel1] - AssertionError: Connecting to 158.177.128.14:6016\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[ecdsap256-bikel3] - AssertionError: Connecting to 158.177.128.14:6002\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[ecdsap256-p384_bikel3] - AssertionError: Connecting to 158.177.128.14:6018\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-x25519] - KeyError: 'X25519'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-frodo976aes] - KeyError: 'frodo976aes'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-p384_frodo976aes] - KeyError: 'p384_frodo976aes'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-frodo976shake] - KeyError: 'frodo976shake'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-p384_frodo976shake] - KeyError: 'p384_frodo976shake'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-frodo1344aes] - KeyError: 'frodo1344aes'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-p521_frodo1344aes] - KeyError: 'p521_frodo1344aes'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-frodo1344shake] - KeyError: 'frodo1344shake'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-p521_frodo1344shake] - KeyError: 'p521_frodo1344shake'\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-bikel1] - AssertionError: Connecting to 158.177.128.14:6087\r\nFAILED oqs_scripts/test_with_interop_server.py::test_sig_kex_pair[dilithium2-p256_bikel1] - AssertionError: Connecting to 158.177.128.14:6092\r\n```\r\n\r\nThis looks like either an issue with a mismatched liboqs/boringssl paring or my other thought was there may be something with the OIDs or NIDs, but I have not executed the objects.go script.\r\nIf anyone could point me in the direction of what is wrong I can correct it, but I'm just at a loss at the moment on where my issue lies.\r\n\r\nAs the title implies, I have done the same setup for Mac Intel, and I am seeing similar test results on that platform as well. \r\n\r\nAny help is very much appreciated.\r\n\r\nThanks!\r\n\r\n-Wyatt\r\n\r\n\r\n","reactions":{"url":"https://api.github.com/repos/open-quantum-safe/boringssl/issues/102/reactions","total_count":0,"+1":0,"-1":0,"laugh":0,"hooray":0,"confused":0,"heart":0,"rocket":0,"eyes":0},"timeline_url":"https://api.github.com/repos/open-quantum-safe/boringssl/issues/102/timeline","performed_via_github_app":null,"state_reason":"completed"}},"public":true,"created_at":"2023-12-04T07:24:55Z","org":{"id":20689385,"login":"open-quantum-safe","gravatar_id":"","url":"https://api.github.com/orgs/open-quantum-safe","avatar_url":"https://avatars.githubusercontent.com/u/20689385?"}}
